From D.Marcinkiewicz@elka.pw.edu.pl Thu Jul 28 09:29:30 2005
Date: Thu, 28 Jul 2005 18:06:13 +0200
From: "Dariusz M." <D.Marcinkiewicz@elka.pw.edu.pl>
To: greg@kroah.com
Subject: USB: pl2303 driver, makes pl2303HX chip work correctly
Message-ID: <20050728160613.GA26735@mion.elka.pw.edu.pl>

This trivial patch makes pl2303 driver work correctly with pl2303HX chip.
Apparently some bug in HX version of pl2303 makes the chip loose some
transmitted bytes or stop working at all after reception of
USB_REQ_CLEAR_FEATURE mesage. Logs generated by UsbSnoop application reveal
that windows driver does not send this type of messages to the converter.

From: "Dariusz M." <D.Marcinkiewicz@elka.pw.edu.pl>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/serial/pl2303.c |    6 ++++--
 1 files changed, 4 insertions(+), 2 deletions(-)

--- gregkh-2.6.orig/drivers/usb/serial/pl2303.c	2005-06-17 12:48:29.000000000 -0700
+++ gregkh-2.6/drivers/usb/serial/pl2303.c	2005-08-08 15:29:04.000000000 -0700
@@ -538,8 +538,10 @@
 
 	dbg("%s -  port %d", __FUNCTION__, port->number);
 
-	usb_clear_halt(serial->dev, port->write_urb->pipe);
-	usb_clear_halt(serial->dev, port->read_urb->pipe);
+	if (priv->type != HX) {
+		usb_clear_halt(serial->dev, port->write_urb->pipe);
+		usb_clear_halt(serial->dev, port->read_urb->pipe);
+	}
 
 	buf = kmalloc(10, GFP_KERNEL);
 	if (buf==NULL)
